src/client/camera.cpp

```diff
@@ -46,6 +46,242 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #define WIELDMESH_AMPLITUDE_X 7.0f
 #define WIELDMESH_AMPLITUDE_Y 10.0f
 
+// Returns the fractional part of x
+inline f32 my_modf(f32 x)
+{
+	f32 dummy;
+	return modff(x, &dummy);
+}
+
+WieldNode::WieldNode(HandIndex index, Client *client, scene::ISceneManager *mgr) :
+	m_index(index),
+	m_direction(index == MAINHAND ? +1 : -1),
+	m_client(client),
+	m_meshnode(new WieldMeshSceneNode(mgr, -1, false)),
+	m_player_light_color(0xFFFFFFFF)
+{
+	m_meshnode->setItem(ItemStack(), m_client);
+	m_meshnode->drop(); // mgr grabbed it
+}
+
+int WieldNode::getDirection()
+{
+	return g_settings->getBool("mirror_hands") ? -m_direction : m_direction;
+}
+
+void WieldNode::step(f32 dtime)
+{
+	bool was_under_zero = m_change_timer < 0;
+	m_change_timer = MYMIN(m_change_timer + dtime, 0.125);
+
+	if (m_change_timer >= 0 && was_under_zero) {
+		m_meshnode->setItem(m_item_next, m_client);
+		m_meshnode->setNodeLightColor(m_player_light_color);
+	}
+
+	if (m_digging_button == -1)
+		return;
+
+	f32 offset = dtime * 3.5f;
+	float m_digging_anim_was = m_digging_anim;
+	m_digging_anim += offset;
+	if (m_digging_anim >= 1)
+	{
+		m_digging_anim = 0;
+		m_digging_button = -1;
+	}
+	float lim = 0.15;
+	if(m_digging_anim_was < lim && m_digging_anim >= lim)
+	{
+		if (m_digging_button == 0) {
+			m_client->getEventManager()->put(new SimpleTriggerEvent(MtEvent::CAMERA_PUNCH_LEFT));
+		} else if(m_digging_button == 1) {
+			m_client->getEventManager()->put(new SimpleTriggerEvent(MtEvent::CAMERA_PUNCH_RIGHT));
+		}
+	}
+}
+
+static inline v2f dir(const v2f &pos_dist)
```

Comment:
very unimportant but IMO the name is too generic
consider `wield_calculate_dir`